samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -156,6 +161,42 @@ public void subscribeToJobModelVersionChange(IZkDataListener dataListener) {
   }
 
   /**
+   * publishes new job model into ZK
+   * @param jobModelVersion  version of the jobModeL to publish
+   * @param jobModel jobModel to publish
+   */
+  public void publishNewJobModel(String jobModelVersion, JobModel jobModel) {
+    try {
+      // This call should FAIL if the node already exists.
```

Comment:
nit: move this to javadoc for the function as it defines the behavior of the function